test_helpers/example_test.go

```diff
@@ -0,0 +1,36 @@
+package test_helpers_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tarantool/go-tarantool/v2"
+	"github.com/tarantool/go-tarantool/v2/test_helpers"
+)
+
+func TestExampleMockDoer(t *testing.T) {
+	mockDoer := test_helpers.NewMockDoer(t,
+		test_helpers.NewMockResponse(t, []interface{}{"some data"}),
```

Comment:
So requests in mock are identified by their position? Is it hard to implement some "if I had received SELECT for `customers` with `id: 1`, return `{1, 'customer1'}`" mock?

Reply:
In this implementation responses will be send in the given order. But it is easy to implement another `Doer`, so it will solve your task. Something like
```go
type AnswerMockDoer struct {
    ...
}

func (doer *AnswerMockDoer) Do(req tarantool.Request) *tarantool.Future {
    if req.Type() == IPROTO_SELECT {
        // Get the id and create a response
        fut.SetResponse(respHeader, respData)
        return fut
    } else {
        // Do something else
    }
    // ...
}
```